demos/interactive_face_detection_demo/cpp/detectors.cpp

```diff
@@ -306,7 +306,7 @@ HeadPoseDetection::HeadPoseDetection(const std::string &pathToModel,
 
 void HeadPoseDetection::submitRequest()  {
     if (!enquedFaces) return;
-    request.set_input_tensor(ov::Tensor{request.get_input_tensor(), {0, 0, 0, 0}, {enquedFaces, inShape[1], inShape[2], inShape[3]}});
+    request.set_input_tensor(ov::Tensor{request.get_input_tensor(), {0, 0, 0, 0}, {enquedFaces + 1, inShape[1], inShape[2], inShape[3]}});
```

Comment:
Do you have an explanation why the behavior is different?
You can reopen the ticket 78597

Reply:
in first case we start from 0, not from enquedFaces. From 0 to  enquedFaces we already have batch size of enquedFaces. From enquedFaces to enquedFaces batch would be 0, that's why we need to do +1. Does it make sense? Because I am starting to think that I misunderstood something.

Reply:
> suggests that the slice is described as [begin, end), similar to c++ iterators.

I see it that way: 
[0, enquedFaces] = 0, 1, 2, ..., enquedFaces -1
and that gives you `enquedFaces` elements it total.
while [enquedFaces, enquedFaces] gives you 0 elements.

Reply:
[Here ](https://github.com/openvinotoolkit/openvino/blob/50dffb80bb955aaf58b8e132f1f139043e201bfb/src/inference/src/ie_layouts.cpp#L532) slices are calculated as `[begin[i], end[i] - begin[i]]` - start index `begin[i]` and size `end[i] - begin[i]`. Size will be part of the shape for resulted roi.  

> Slice is described as [begin, end), similar to c++ iterators.

That's right, and in case of begin == end, we would have 0 elements between them. If interval is [begin, begin + 1) == [begin, begin] - 1 element. End if we continue, [0, enquedFaces+1) == [0, enquedFaces] - that's enquedFaces + 1 elements between them. The misunderstanding comes from mixing indexing and actual number of elements. Indexing suggests that for [begin, end) the number of elements would be end - begin.

Reply:
Got it. Thanks